app/src/main/java/org/mozilla/fenix/library/downloads/DownloadFragment.kt

```diff
@@ -73,12 +93,58 @@ class DownloadFragment : LibraryPageFragment<DownloadItem>(), UserInteractionHan
 
     override val selectedItems get() = downloadStore.state.mode.selectedItems
 
+    private fun invalidateOptionsMenu() {
+        activity?.invalidateOptionsMenu()
+    }
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+        setHasOptionsMenu(true)
+    }
 
-        requireComponents.analytics.metrics.track(Event.HistoryOpened)
+    private fun displayDeleteAll() {
+        activity?.let { activity ->
+            AlertDialog.Builder(activity).apply {
+                setMessage(R.string.download_delete_all_dialog)
+                setNegativeButton(R.string.delete_browsing_data_prompt_cancel) { dialog: DialogInterface, _ ->
+                    dialog.cancel()
+                }
+                setPositiveButton(R.string.delete_browsing_data_prompt_allow) { dialog: DialogInterface, _ ->
+                    downloadStore.dispatch(DownloadFragmentAction.EnterDeletionMode)
```

Comment:
Do we have to enter the deletion mode here?

It might be easier to just cancel any pending deletion operations and then call `removeAllDownloads()`.